### Introdction

Realign reads to transcriptome assembly.
Capture and count all reads that map our transcripts.
Count all properly paired mapped reads and the reads where just one read pair maps.

### Using this approach https://github.com/trinityrnaseq/trinityrnaseq/wiki/RNA-Seq-Read-Representation-by-Trinity-Assembly

Didn't use the -SS_lib_type flag for de_novo_1, because it was also not used for the assembly.

But used it for de_novo_2.

* Results




### Using alvars mapping_filtering.sh script



* Results

